src/ucp/rndv/rndv.c

```diff
@@ -440,12 +440,39 @@ static void ucp_rndv_req_send_rtr(ucp_request_t *rndv_req, ucp_request_t *rreq,
     ucp_request_send(rndv_req);
 }
 
+static uint8_t ucp_rndv_get_rkey_index(ucp_request_t *rndv_req, ucp_rkey_h rkey,
+                                       ucp_lane_index_t lane)
+{
+    ucp_ep_h ep                 = rndv_req->send.ep;
+    ucp_context_h context       = ep->worker->context;
+    ucp_ep_config_t *ep_config  = ucp_ep_config(ep);
+    ucp_md_index_t md_index     = ep_config->md_index[lane];
+    uct_md_attr_v2_t *md_attr   = &context->tl_mds[md_index].attr;
```

Comment:
should not use md_index as array index before we check it's not UCP_NULL_RESOURCE
even if we don't use the pointer it is a bad practice and dangerous for future code changes